### ISSUE TITLE
Needs loose description Luhn algorithm and potential applications for readme

### ISSUE THREAD

**Issue by voronoipotato** [gtbuchanan/Luhny#153]
Firstly this sounds pretty cool! I was thinking we could yoink some text from wikipedia with an acknowledgement. This is a bit wordy, but I think the relevant parts are, "This is a few places where you might need it" "The algorithm is public domain" "It is not designed for cryptographic security but rather to prevent accidental entry".  

Thanks for your contribution to humanity, every little bit adds up. 

---

From wikipedia 
### The Luhn algorithm is used in a variety of systems, including:

[Credit card numbers](https://en.wikipedia.org/wiki/Payment_card_number)
[IMEI numbers](https://en.wikipedia.org/wiki/International_Mobile_Equipment_Identity)
[National Provider Identifier numbers](https://en.wikipedia.org/wiki/National_Provider_Identifier) in the United States
[Canadian](https://en.wikipedia.org/wiki/Canada) [social insurance numbers](https://en.wikipedia.org/wiki/Social_insurance_number)
[Israeli](https://en.wikipedia.org/wiki/Israeli_identity_card) ID numbers
[South African](https://en.wikipedia.org/wiki/South_Africa) ID numbers
[South African](https://en.wikipedia.org/wiki/South_Africa) Tax reference numbers
[Swedish](https://en.wikipedia.org/wiki/Sweden) [national identification numbers](https://en.wikipedia.org/wiki/National_identification_number)
[Swedish](https://en.wikipedia.org/wiki/Sweden) Corporate Identity Numbers (OrgNr)
[Greek](https://en.wikipedia.org/wiki/Greece) Social Security Numbers (ΑΜΚΑ)
[ICCID](https://en.wikipedia.org/wiki/SIM_card#ICCID) of SIM cards
[European patent](https://en.wikipedia.org/wiki/European_Patent_Convention) application numbers
Survey codes appearing on [McDonald's](https://en.wikipedia.org/wiki/File:McDonalds_Receipt_Luhn_Algorithm.png), [Taco Bell](https://en.wikipedia.org/wiki/File:Taco_Bell_Receipt_Luhn_Algorithm.png), and [Tractor Supply Co.](https://en.wikipedia.org/wiki/File:Tractor_Supply_Receipt_Luhn_Algorithm.png) receipts
[United States Postal Service](https://en.wikipedia.org/wiki/United_States_Postal_Service) package tracking numbers use a modified Luhn algorithm[[4]](https://en.wikipedia.org/wiki/Luhn_algorithm#cite_note-4)
Italian VAT numbers ([Partita Iva](https://en.wikipedia.org/wiki/VAT_identification_number#European_Union_VAT_identification_numbers))[[5]](https://en.wikipedia.org/wiki/Luhn_algorithm#cite_note-5)

The Luhn algorithm or Luhn formula, also known as the "[modulus](https://en.wikipedia.org/wiki/Modular_arithmetic) 10" or "mod 10" [algorithm](https://en.wikipedia.org/wiki/Algorithm), named after its creator, [IBM](https://en.wikipedia.org/wiki/IBM) scientist [Hans Peter Luhn](https://en.wikipedia.org/wiki/Hans_Peter_Luhn), is a simple [check digit](https://en.wikipedia.org/wiki/Check_digit) formula used to validate a variety of identification numbers. It is described in [US](https://en.wikipedia.org/wiki/United_States) patent 2950048A, granted on 23 August 1960.[[1]](https://en.wikipedia.org/wiki/Luhn_algorithm#cite_note-US2950048A-1).

The algorithm is in the [public domain](https://en.wikipedia.org/wiki/Public_domain) and is in wide use today. It is specified in [ISO/IEC 7812-1](https://en.wikipedia.org/wiki/ISO/IEC_7812-1).[[2]](https://en.wikipedia.org/wiki/Luhn_algorithm#cite_note-2) It is not intended to be a [cryptographically secure hash function](https://en.wikipedia.org/wiki/Cryptographic_hash_function); it was designed to protect against accidental errors, not malicious attacks. Most credit cards and many government identification numbers use the algorithm as a simple method of distinguishing valid numbers from mistyped or otherwise incorrect numbers.

**Comment by gtbuchanan** [comment]
Thank you for the suggestion! This was mostly an exercise to improve my F# skills, so it is still in its infancy, but I will definitely consider adding more documentation as I clean things up.

**Comment by voronoipotato** [comment]
Yeah I figured even if people don't leverage it yet, it would still be good for teaching :).
